ophyd/v2/_aioca.py

```diff
@@ -211,10 +219,16 @@ async def get_value(self) -> T:
         value = await self._caget(FORMAT_RAW)
         return self.converter.value(value)
 
-    def monitor_reading_value(self, callback: ReadingValueCallback[T]) -> Monitor:
-        return camonitor(
-            self.read_pv,
-            lambda v: callback(self.converter.reading(v), self.converter.value(v)),
-            datatype=self.converter.read_dbr,
-            format=FORMAT_TIME,
-        )
+    def set_callback(self, callback: Optional[ReadingValueCallback[T]]) -> None:
+        if callback:
+            assert (
+                not self.subscription
+            ), "Cannot set a callback when one is already set"
+            self.subscription = camonitor(
+                self.read_pv,
+                lambda v: callback(self.converter.reading(v), self.converter.value(v)),
+                datatype=self.converter.read_dbr,
+                format=FORMAT_TIME,
+            )
+        else:
+            self.subscription = None
```

Comment:
Need to close the subscription first
```suggestion
            if self.subscription:
                self.subscription.close()
            self.subscription = None
```